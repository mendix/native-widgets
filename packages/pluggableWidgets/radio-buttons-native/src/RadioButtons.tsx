import { createElement, useCallback, ReactElement } from "react";
import { Text, View } from "react-native";

import { mergeNativeStyles } from "@mendix/pluggable-widgets-tools";
import { executeAction } from "@mendix/piw-utils-internal";

import { RadioButton } from "./components/RadioButton";
import { RadioButtonsProps } from "../typings/RadioButtonsProps";
import { defaultRadioButtonsStyle, RadioButtonsStyle } from "./ui/Styles";

export type props = RadioButtonsProps<RadioButtonsStyle>;

export function RadioButtons({
    enum: { setValue, universe, value, formatter, readOnly, validation },
    orientation,
    style,
    onChange,
    name,
    label,
    showLabel,
    accessible,
    screenReaderHint
}: props): ReactElement {
    const styles = mergeNativeStyles(defaultRadioButtonsStyle, style);
    const onSelect = useCallback(
        (selectedValue: string) => {
            if (selectedValue === value || readOnly) {
                return;
            }
            setValue(selectedValue);
            executeAction(onChange);
        },
        [onChange, readOnly, setValue, value]
    );

    return (
        <View accessibilityHint={screenReaderHint?.value} testID={name} style={styles.container}>
            {showLabel && <Text style={styles.labelTextStyle}>{label?.value}</Text>}
            <View style={orientation === "horizontal" && styles.containerHorizontal}>
                {universe?.map(name => (
                    <RadioButton
                        accessible={accessible === "yes"}
                        key={name}
                        active={value === name}
                        onSelect={onSelect}
                        title={formatter.format(name)}
                        styles={styles}
                        name={name}
                        disabled={readOnly}
                        orientation={orientation}
                    />
                ))}
            </View>
            {validation && <Text style={styles.validationMessage}>{validation}</Text>}
        </View>
    );
}
