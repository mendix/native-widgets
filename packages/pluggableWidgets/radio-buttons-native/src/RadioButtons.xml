<?xml version="1.0" encoding="UTF-8" ?>
<widget xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="com.mendix.widget.native.radiobuttons.RadioButtons" pluginWidget="true" needsEntityContext="true" offlineCapable="true" supportedPlatform="Native" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../node_modules/mendix/custom_widget.xsd">
    <name>Radio Buttons</name>
    <description />
    <studioProCategory>Input elements</studioProCategory>
    <studioCategory>Input elements</studioCategory>
    <icon />
    <properties>
        <propertyGroup caption="General">
            <propertyGroup caption="General">
                <property key="orientation" type="enumeration" defaultValue="vertical">
                    <caption>Orientation</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="vertical">Vertical</enumerationValue>
                        <enumerationValue key="horizontal">Horizontal</enumerationValue>
                    </enumerationValues>
                </property>
            </propertyGroup>
            <propertyGroup caption="Data Source">
                <property key="enum" type="attribute">
                    <caption>Attribute</caption>
                    <description />
                    <attributeTypes>
                        <attributeType name="Enum" />
                    </attributeTypes>
                </property>
            </propertyGroup>
            <propertyGroup caption="Label">
                <property key="showLabel" type="boolean" defaultValue="false">
                    <caption>Show label</caption>
                    <description />
                </property>
                <property key="label" type="textTemplate" required="false">
                    <caption>Label caption</caption>
                    <description />
                </property>
            </propertyGroup>
            <propertyGroup caption="Editability">
                <systemProperty key="Editability" />
            </propertyGroup>
            <propertyGroup caption="Visibility">
                <systemProperty key="Visibility" />
            </propertyGroup>
            <propertyGroup caption="Common">
                <systemProperty key="Name" />
            </propertyGroup>
            <propertyGroup caption="Accessibilty">
                <property key="accessible" type="enumeration" defaultValue="yes">
                    <caption>Accessible</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="screenReaderCaption" type="textTemplate" required="false">
                    <caption>Screen reader caption</caption>
                    <description />
                </property>
                <property key="screenReaderHint" type="textTemplate" required="false">
                    <caption>Screen reader hint</caption>
                    <description />
                </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Events">
            <propertyGroup caption="Events">
                <property key="onChange" type="action" required="false">
                    <caption>On change</caption>
                    <description />
                </property>
            </propertyGroup>
        </propertyGroup>
    </properties>
</widget>
