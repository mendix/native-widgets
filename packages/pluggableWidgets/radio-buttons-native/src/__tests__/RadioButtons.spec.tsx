import { createElement } from "react";
import { actionValue, dynamicValue, EditableValueBuilder } from "@mendix/piw-utils-internal";
import { RadioButtons, props } from "../RadioButtons";
import { fireEvent, render } from "@testing-library/react-native";

const enumValues = ["firstItem", "secondItem"];

function getEnumValueTestId(index: number): string {
    return `radio-button-${enumValues[index]}`;
}

describe("Radio buttons", () => {
    let defaultProps: props;
    beforeEach(() => {
        defaultProps = {
            enum: new EditableValueBuilder<string>()
                .withUniverse(...enumValues)
                .withValue(enumValues[0])
                .build(),
            name: "radio-buttons-test",
            accessible: "yes",
            orientation: "vertical",
            style: [],
            onChange: actionValue(),
            label: dynamicValue<string>("Label", false),
            showLabel: true
        };
    });
    it("render vertical radio buttons correctly", () => {
        const component = render(<RadioButtons {...defaultProps} orientation="vertical" />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("render horizontal radio buttons correctly", () => {
        const component = render(<RadioButtons {...defaultProps} orientation="horizontal" />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("render correctly with show label false", () => {
        const component = render(<RadioButtons {...defaultProps} showLabel={false} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("render correctly without label", () => {
        const component = render(<RadioButtons {...defaultProps} label={undefined} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("works correctly without options", () => {
        defaultProps.enum = new EditableValueBuilder<string>().build();
        const component = render(<RadioButtons {...defaultProps} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("shows validations correctly", () => {
        defaultProps.enum = new EditableValueBuilder<string>().withValidation("error message").build();
        const component = render(<RadioButtons {...defaultProps} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("render disabled radio buttons correctly", () => {
        defaultProps.enum = new EditableValueBuilder<string>()
            .withUniverse(...enumValues)
            .isReadOnly()
            .build();
        const component = render(<RadioButtons {...defaultProps} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("call set value when active radio button changes", () => {
        const component = render(<RadioButtons {...defaultProps} />);
        fireEvent.press(component.getByTestId(getEnumValueTestId(1)));
        expect(defaultProps.enum.setValue).toHaveBeenCalledWith(enumValues[1]);
    });
    it("doesn't call set value when active radio button pressed", () => {
        const component = render(<RadioButtons {...defaultProps} />);
        fireEvent.press(component.getByTestId(getEnumValueTestId(0)));
        expect(defaultProps.enum.setValue).not.toHaveBeenCalled();
    });
    it("value changes when not active radio button pressed", () => {
        const component = render(<RadioButtons {...defaultProps} />);
        fireEvent.press(component.getByTestId(getEnumValueTestId(1)));
        expect(defaultProps.enum.value).toBe(enumValues[1]);
    });
    it("on change event triggered when active radio button changes", () => {
        const component = render(<RadioButtons {...defaultProps} />);
        fireEvent.press(component.getByTestId(getEnumValueTestId(1)));
        expect(defaultProps.onChange?.execute).toHaveBeenCalled();
    });
});
