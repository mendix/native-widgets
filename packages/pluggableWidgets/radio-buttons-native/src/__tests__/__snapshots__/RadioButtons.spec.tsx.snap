// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio buttons render correctly with show label false 1`] = `
<View
  style={{}}
  testID="radio-buttons-test"
>
  <View
    style={false}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          false,
        ]
      }
      testID="radio-button-firstItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      >
        <View
          style={
            {
              "backgroundColor": "#264AE5",
              "borderRadius": 4,
              "height": 8,
              "width": 8,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted firstItem
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          false,
        ]
      }
      testID="radio-button-secondItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted secondItem
      </Text>
    </View>
  </View>
</View>
`;

exports[`Radio buttons render correctly without label 1`] = `
<View
  style={{}}
  testID="radio-buttons-test"
>
  <Text
    style={
      {
        "color": "#0A1326",
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 8,
      }
    }
  />
  <View
    style={false}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          false,
        ]
      }
      testID="radio-button-firstItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      >
        <View
          style={
            {
              "backgroundColor": "#264AE5",
              "borderRadius": 4,
              "height": 8,
              "width": 8,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted firstItem
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          false,
        ]
      }
      testID="radio-button-secondItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted secondItem
      </Text>
    </View>
  </View>
</View>
`;

exports[`Radio buttons render disabled radio buttons correctly 1`] = `
<View
  style={{}}
  testID="radio-buttons-test"
>
  <Text
    style={
      {
        "color": "#0A1326",
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <View
    style={false}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          {
            "opacity": 0.5,
          },
        ]
      }
      testID="radio-button-firstItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted firstItem
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          {
            "opacity": 0.5,
          },
        ]
      }
      testID="radio-button-secondItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted secondItem
      </Text>
    </View>
  </View>
</View>
`;

exports[`Radio buttons render horizontal radio buttons correctly 1`] = `
<View
  style={{}}
  testID="radio-buttons-test"
>
  <Text
    style={
      {
        "color": "#0A1326",
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          {
            "marginBottom": 0,
            "marginEnd": 8,
          },
          false,
        ]
      }
      testID="radio-button-firstItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      >
        <View
          style={
            {
              "backgroundColor": "#264AE5",
              "borderRadius": 4,
              "height": 8,
              "width": 8,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted firstItem
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          {
            "marginBottom": 0,
            "marginEnd": 8,
          },
          false,
        ]
      }
      testID="radio-button-secondItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted secondItem
      </Text>
    </View>
  </View>
</View>
`;

exports[`Radio buttons render vertical radio buttons correctly 1`] = `
<View
  style={{}}
  testID="radio-buttons-test"
>
  <Text
    style={
      {
        "color": "#0A1326",
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <View
    style={false}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          false,
        ]
      }
      testID="radio-button-firstItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      >
        <View
          style={
            {
              "backgroundColor": "#264AE5",
              "borderRadius": 4,
              "height": 8,
              "width": 8,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted firstItem
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 8,
          },
          false,
          false,
        ]
      }
      testID="radio-button-secondItem"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#CED0D3",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginEnd": 4,
              "width": 16,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#0A1326",
            "fontSize": 12,
            "lineHeight": 18,
          }
        }
      >
        Formatted secondItem
      </Text>
    </View>
  </View>
</View>
`;

exports[`Radio buttons shows validations correctly 1`] = `
<View
  style={{}}
  testID="radio-buttons-test"
>
  <Text
    style={
      {
        "color": "#0A1326",
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <View
    style={false}
  />
  <Text
    style={
      {
        "color": "#E33F4E",
        "fontSize": 12,
      }
    }
  >
    error message
  </Text>
</View>
`;

exports[`Radio buttons works correctly without options 1`] = `
<View
  style={{}}
  testID="radio-buttons-test"
>
  <Text
    style={
      {
        "color": "#0A1326",
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <View
    style={false}
  />
</View>
`;
